hail/python/hail/__init__.py

```diff
@@ -1,13 +1,23 @@
+import pkg_resources
+import sys
+import asyncio
 import nest_asyncio
-nest_asyncio.apply()
-
-import pkg_resources  # noqa: E402
-import sys  # noqa: E402
 
 if sys.version_info < (3, 6):
     raise EnvironmentError('Hail requires Python 3.6 or later, found {}.{}'.format(
         sys.version_info.major, sys.version_info.minor))
 
+if sys.version_info[:2] == (3, 6):
+    if asyncio._get_running_loop() is not None:
+        nest_asyncio.apply()
+else:
+    try:
+        asyncio.get_running_loop()
+        nest_asyncio.apply()
+    except RuntimeError as err:
+        assert 'no running event loop' in err.args[0]
```

Comment:
I had to add this nonsense to avoid some pytest errors I saw in the service tests.

I think pytest imports python modules from a secondary thread but executes them from the main thread. If `nest_asyncio` is called when a loop isn't running, it tries to create a loop. Unfortunately, `asyncio` forbids loops outside the main thread. Therefore, we get an error. Luckily, pytest does not need nest_asyncio because it doesn't start an asyncio event loop.

Regarding the Python 3.6 underscore method: 3.6 doesn't have an officially supported way to do this. That's why they added `get_running_loop` in 3.7.

cc: @tpoterba